azure-pipelines.yml

```diff
@@ -51,204 +54,23 @@ jobs:
-    displayName: 'Install pip'
-  - script: sudo python3 -m pip install -r InstallerGUI/requirements.txt
-    displayName: 'Install requirements'
-
-  - script: |
-      mono --version
-      sudo cat /etc/apt/sources.list.d/mono-official-stable.list
-      sudo apt-get remove mono-complete
-      sudo apt autoremove
-      sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
-      sudo apt install apt-transport-https ca-certificates
-      echo "deb https://download.mono-project.com/repo/ubuntu wheezy/snapshots/4.6.2 main" | sudo tee /etc/apt/sources.list.d/mono-official-stable.list
-      sudo apt update
-      sudo apt install mono-devel
-      mono --version
-    displayName: 'Install Mono 4.6.2'
-
-  - script: chmod +x InstallerGUI/mkbundle.sh
-    displayName: 'Make mkbundle.sh executable'
-  - script: pyinstaller -y pyinstaller.spec
-    displayName: 'Create installer'
-    workingDirectory: $(Build.SourcesDirectory)/InstallerGUI
-
-  - task: CopyFiles@2
-    inputs:
-      sourceFolder: $(system.defaultworkingdirectory)/InstallerGUI/dist/
-      contents: '**'
-      targetFolder: $(Build.ArtifactStagingDirectory)
-
-  - bash: mv speedrun-timer-installer_linux.x64 "$(installerBaseFileName)_linux.x64"
-    workingDirectory: $(Build.ArtifactStagingDirectory)
-    displayName: Rename installer
-
-  - task: PublishBuildArtifacts@1
-    displayName: Publish linux installer artifact
-    inputs:
-      pathtoPublish: '$(Build.ArtifactStagingDirectory)'
-      artifactName: 'installer-linux'
-      publishLocation: 'Container'
-
-  - script: ls -lR $(Build.ArtifactStagingDirectory)
-
-- job: Deploy
-  condition: startsWith(variables['Build.SourceBranch'], 'refs/tags/')
-  dependsOn:
-    - WindowsInstaller
-    - LinuxInstaller
-    - MacInstaller
-  pool:
-    vmImage: 'VS2017-Win2016'
-  steps:
-
-  - task: DownloadBuildArtifacts@0
-    displayName: Download windows installer artifact
-    inputs:
-      artifactName: 'installer-windows'
-      downloadPath: '$(System.ArtifactsDirectory)'
-
-  - task: DownloadBuildArtifacts@0
-    displayName: Download macOs installer artifact
-    inputs:
-      artifactName: 'installer-macOs'
-      downloadPath: '$(System.ArtifactsDirectory)'
-
-  - task: DownloadBuildArtifacts@0
-    displayName: Download linux installer artifact
-    inputs:
-      artifactName: 'installer-linux'
-      downloadPath: '$(System.ArtifactsDirectory)'
+  - script: dotnet publish -r linux-x64 -c Release -f netcoreapp2.1 -o $(Build.ArtifactStagingDirectory)/linux SpeedrunModInstaller.Avalonia/SpeedrunModInstaller.Avalonia.csproj
+    condition: startsWith(variables['Build.SourceBranch'], 'refs/tags/')
+    displayName: 'Publish linux installer'
 
   - bash: ls -lR "$(System.ArtifactsDirectory)"
 
   - task: GitHubRelease@0
+    condition: startsWith(variables['Build.SourceBranch'], 'refs/tags/')
+    enabled: 'false' # not fixed yet
```

Comment:
Things still needs to be zipped up and prepared for deploy

Reply:
Files are now zipped, but could do with some improvements for usability. (The installer is one specific file in a folder full of other files.)